### PR TITLE
redo setDefaultLibrary()

### PR DESCRIPTION
**Pull request by WalterBright**
1. changed name to plural
2. added Ddoc comment
3. remove unused parameter
4. replace use of global with parameter

### CODE REVIEW

**Comment by WalterBright** [comment]
5. add `pure` and `@safe`

File: compiler/src/dmd/mars.d [hunk 1]
```diff
@@ -343,35 +343,40 @@ const(char)[] parse_conf_arg(Strings* args)
...
+pure @safe
+void setDefaultLibraries(const ref Target target, ref const(char)[] defaultlibname, ref const(char)[] debuglibname)
 {
-    if (driverParams.defaultlibname is null)
+    if (defaultlibname is null)
     {
         if (target.os == Target.OS.Windows)
         {
-            driverParams.defaultlibname = target.isX86_64 ? "phobos64" : "phobos32mscoff";
+            defaultlibname = target.isX86_64 ? "phobos64" : "phobos32mscoff";
```
Comment (thewilsonator) [inline compiler/src/dmd/mars.d:358]: at some point I'd love to rename `phobos32mscoff` to `phobos32`, but I could never figure out the correct build system invocation corrections.

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/mars.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/mars.d:358]: Add a bugzilla issue for that!

Comment (thewilsonator) [reply compiler/src/dmd/mars.d:358]: Done: https://issues.dlang.org/show_bug.cgi?id=24576

**Pull request merged**